docs/standard/io/Buffers.md

```diff
@@ -0,0 +1,401 @@
+# System.Buffers
+
+- **Writing**
+    - [IBufferWriter\<T\>](#ibufferwritert)
+       - [Gotchas](#gotchas)
+- **Reading**
+    - [ReadOnlySequence\<T\>](#readonlysequencet)
+       - [Accessing data](#accessing-data)
+       - [Gotchas](#gotchas-1)
+    - [SequenceReader\<T\>](#sequencereadert)
+       - [Accessing data](#accessing-data-1)
+       - [Gotchas](#gotchas-2)
+   
+## [IBufferWriter\<T\>](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.ibufferwriter-1?view=netstandard-2.1)
+
+`IBufferWriter<T>` is a contract for synchronous buffered writing. At the lowest level the interface is extremely simple and allows you to get access to a `Memory<T>` or `Span<T>`, write to it and say how many `T` items were written.
+
+```C#
+void WriteHello(IBufferWriter<byte> writer)
+{
+    // Request at least 5 bytes
+    Span<byte> span = writer.GetSpan(5);
+    ReadOnlySpan<char> helloSpan = "Hello".AsSpan();
+    int written = Encoding.ASCII.GetBytes(helloSpan, span);
+    
+    // Tell the writer how many bytes we wrote
+    writer.Advance(written);
+}
+```
+
+The preceding method requests a buffer of at least 5 bytes from the `IBufferWriter<byte>` using `GetSpan(5)` then writes bytes for the ASCII string "Hello" to the returned `Span<byte>`. It then calls `Advance(written)` to indicate how many bytes were written to the buffer. 
+
+This method of writing uses the `Memory<T>`/`Span<T>` buffer provided by the `IBufferWriter<T>`, but you can also use the [`Write`](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.buffersextensions.write?view=netstandard-2.1) extension method to copy an existing buffer to the `IBufferWriter<T>`. `Write` does the work of calling `GetSpan`/`Advance` as appropriate, so there's no need to call `Advance` after writing.
```

Comment:
We should have links to the API doc for GetSpan/GetMemory/Advance. Ex: https://docs.microsoft.com/en-us/dotnet/api/system.buffers.ibufferwriter-1.getmemory?view=netstandard-2.1#System_Buffers_IBufferWriter_1_GetMemory_System_Int32_

Reply:
> PLEASE HOLD COMMENTS UNTIL IT'S READY FOR REVIEW

😆

Reply:
Yeah, I stopped as soon as I saw that. Oops!

Reply:
NP. This is taking a back seat to the ASP.NET Core 3.0 release so I'll be slow doing updates. At some point I'll need one of you to get the bulk of the pipe code into a working project on a Git repo. That way I can migrate the code snippets to this doc and not have embedded snippets.